File: YocoInterviewTake5/TablesiOS_demo5/Models/BusinessCategory.cs
using System;
using SQLite;

namespace YocoInterviewTake5
{
	public class BusinessCategory 
	{
		string iD;
		[PrimaryKey]
		public string ID {
			get {
				return iD;
			}
			set {
				iD = value;
			}
		}

		string category;
		public string Category {
			get {
				return category;
			}
			set {
				category = value;
			}
		}

		string name;
		public string Name {
			get {
				return name;
			}
			set {
				name = value;
			}
		}
	}
}

File: YocoInterViewObjCMain/YocoInterviewObjC/YocoInterviewObjC/BusinessCategory.m
//
//  BusinessCategory.m
//  YocoInterviewObjC
//
//  Created by Rohan Jansen on 2015/02/12.
//  Copyright (c) 2015 Rohan Jansen. All rights reserved.
//

#import "BusinessCategory.h"

@implementation BusinessCategory

@synthesize Id;
@synthesize Category;
@synthesize Name;

#pragma - Encoder

- (id)initWithCoder:(NSCoder *)decoder {
    self = [super init];
    if (!self) {
        return nil;
    }
    
    self.Id = [decoder decodeObjectForKey:@"Id"];
    self.Category = [decoder decodeObjectForKey:@"Category"];
    self.Name = [decoder decodeObjectForKey:@"Name"];
    
    return self;
}

- (void)encodeWithCoder:(NSCoder *)encoder {
    [encoder encodeObject:self.Id forKey:@"Id"];
    [encoder encodeObject:self.Category forKey:@"Category"];
    [encoder encodeObject:self.Name forKey:@"Name"];
}

@end
